File: JavaScript/881-Boats to Save People.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    people.sort(function(a,b){return a-b;});
    
    let start = 0, end = people.length-1;
    let res = 0;
    
    while (start <= end){
        res += 1;
        if (people[start] + people[end] <= limit){
            start++;end--;
        }
        else{
            end--;
        }
    }
    return res;
};

File: C++/881-Boats to Save People.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(begin(people), end(people));
        int res = 0;
        int right = people.size()-1, left = 0;

        while (right >= left) {
            if (right > left && people[right] + people[left] <= limit) {
                left++;
            }
            right--;
            res++;
        }

        return res;
    }
};